staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -333,25 +337,48 @@ func ValidateWebhookConfiguration(fldPath *field.Path, c *api.WebhookConfigurati
 	default:
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("connectionInfo", "type"), c.ConnectionInfo, []string{api.AuthorizationWebhookConnectionInfoTypeInCluster, api.AuthorizationWebhookConnectionInfoTypeKubeConfigFile}))
 	}
+	if len(c.MatchConditions) > 64 {
+		allErrs = append(allErrs, field.TooMany(fldPath.Child("matchConditions"), len(c.MatchConditions), 64))
+	}
+	for i, condition := range c.MatchConditions {
+		if i >= 64 {
+			break
+		}
+		fldPath := fldPath.Child("matchConditions").Index(i).Child("expression")
+		if len(strings.TrimSpace(condition.Expression)) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath, ""))
+		} else {
+			allErrs = append(allErrs, validateMatchConditionsExpression(fldPath, condition.Expression)...)
+		}
+	}
+	return allErrs
+}
 
-	// TODO: Remove this check and ensure that correct validations below for MatchConditions are added
-	// for i, condition := range c.MatchConditions {
-	//	 fldPath := fldPath.Child("matchConditions").Index(i).Child("expression")
-	//	 if len(strings.TrimSpace(condition.Expression)) == 0 {
-	//	     allErrs = append(allErrs, field.Required(fldPath, ""))
-	//	 } else {
-	//		 allErrs = append(allErrs, ValidateWebhookMatchCondition(fldPath, sampleSAR, condition.Expression)...)
-	//	 }
-	// }
-	if len(c.MatchConditions) != 0 {
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("matchConditions"), c.MatchConditions, []string{}))
+func validateMatchConditionsExpression(fldPath *field.Path, expression string) field.ErrorList {
+	allErrors := field.ErrorList{}
+	compiler := authorizationcel.NewCompiler(environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()))
+	authzExpression := &authorizationcel.SubjectAccessReviewMappingCondition{
+		Expression: expression,
 	}
 
-	return allErrs
+	result, _ := compiler.CompileCELExpression(authzExpression)
```

Comment:
1 of the comments I got for the Authn PR was to return the error as part of `CompileCELExpression` instead of storing the error in the `CompilationResult`. @enj should we do the same here?